google/_modules/gke/cluster.tf

```diff
@@ -49,4 +49,13 @@ resource "google_container_cluster" "current" {
       start_time = var.daily_maintenance_window_start_time
     }
   }
+
+  private_cluster_config {
```

Comment:
```
Step #4 - "terraform plan":       + private_cluster_config {
Step #4 - "terraform plan":           + enable_private_nodes   = true # forces replacement
Step #4 - "terraform plan":           + master_ipv4_cidr_block = "10.254.0.0/28" # forces replacement
Step #4 - "terraform plan":           + peering_name           = (known after apply)
Step #4 - "terraform plan":           + private_endpoint       = (known after apply)
Step #4 - "terraform plan":           + public_endpoint        = (known after apply)
Step #4 - "terraform plan":         }
```

Reply:
This is fine, especially since `enable_private_nodes = false` can be used to retain previous behavior. But setting this produces the following error currently:

```
Step #4 - "terraform plan": Error: Provider produced invalid plan
Step #4 - "terraform plan": 
Step #4 - "terraform plan": Provider "registry.terraform.io/-/google" has indicated "requires replacement"
Step #4 - "terraform plan": on module.gke_zero.module.cluster.google_container_cluster.current for a
Step #4 - "terraform plan": non-existent attribute path
Step #4 - "terraform plan": cty.Path{cty.GetAttrStep{Name:"private_cluster_config"},
Step #4 - "terraform plan": cty.IndexStep{Key:cty.NumberIntVal(0)},
Step #4 - "terraform plan": cty.GetAttrStep{Name:"master_ipv4_cidr_block"}}.
Step #4 - "terraform plan": 
Step #4 - "terraform plan": This is a bug in the provider, which should be reported in the provider's own
Step #4 - "terraform plan": issue tracker.
```

Need to report that upstream.

Reply:
Upstream misunderstood this PR as trying to fix the issue and refused to look into it. Very unfortunate.

Reply:
Yeah I saw that, I thought they would reopen once you mentioned it